packages/backend/src/open_payments/payment/incoming/routes.ts

```diff
@@ -71,16 +71,17 @@ async function getIncomingPayment(
   } catch (err) {
     ctx.throw(500, 'Error trying to get incoming payment')
   }
-  if (!incomingPayment) return ctx.throw(404)
+  if (!incomingPayment || !incomingPayment.walletAddress) return ctx.throw(404)
   const connection = deps.connectionService.get(incomingPayment)
   ctx.body = incomingPaymentToBody(
-    ctx.walletAddress,
+    incomingPayment.walletAddress,
     incomingPayment,
     connection
   )
```

Comment:
I'm assuming the connection part is going to be removed in a subsequent PR from @BlairCurrey ?

Reply:
That's correct.

Reply:
This part will actually still be here but with a paired down `connection` type. After talking w/ Max our plan was to leave the connection service because we still need to get credentials from the stream server. So we will just have `ConnectionService.get` that returns something like this (and getting rid of the current `Connection` model):
```ts
{
  ilpAddress: IlpAddress
  sharedSecret: Buffer
}
```
So internally we still have some concept of connections but they are basically just the stream credentials. I could probably see a case for putting this somewhere in `IncomingPaymentsService` or the model but for now this is the easier/faster refactor.